src/views/app.tsx

```diff
@@ -1,106 +1,39 @@
-import BlankScreen from "ui/components/shared/BlankScreen";
 import App from "ui/components/App";
 import { bootstrapApp } from "ui/setup";
-import "image/image.css";
 
-const url = new URL(window.location.href);
-const recordingId = url.searchParams.get("id");
+import "image/image.css";
 
-const Upload = React.lazy(() => import("views/upload"));
+const Recording = React.lazy(() => import("./recording"));
 const Account = React.lazy(() => import("ui/components/Account"));
-const DevTools = React.lazy(() => import("ui/components/DevTools"));
 
 bootstrapApp();
 
-function PageSwitch({ setExpectedError, setWorkspaceId }: PropsFromRedux) {
-  const { isAuthenticated, user } = useAuth0();
-  const { userSettings, loading: settingsLoading } = useGetUserSettings();
-  const {
-    initialized: recordingInitialized,
-    loading: initializedLoading,
-    graphQLError,
-  } = useIsRecordingInitialized(recordingId);
-  const { userId: ownerId, loading: ownerIdLoading } = useGetRecordingOwnerUserId(recordingId);
-
-  useEffect(() => {
-    if (userSettings) {
-      setWorkspaceId(userSettings.defaultWorkspaceId);
-    }
-  }, [userSettings]);
-
-  useEffect(() => {
-    if (isAuthenticated) {
-      bootIntercom({ email: user.email });
-    }
-  }, [isAuthenticated]);
-
-  useEffect(() => {
-    if (recordingId) {
-      if (!validateUUID(recordingId)) {
-        return setExpectedError({
-          message: "Invalid ID",
-          content: `"${recordingId}" is not a valid recording ID`,
-        });
-      }
-      if (graphQLError) {
-        return setExpectedError({ message: "Error", content: graphQLError });
-      }
-    }
-  }, [graphQLError]);
-
-  if (initializedLoading || ownerIdLoading || settingsLoading) {
-    return null;
-  }
-
-  if (recordingId) {
-    if (!validateUUID(recordingId) || graphQLError) {
-      return <BlankScreen />;
-    }
-
-    // Add a check to make sure the recording has an associated user ID.
-    // We skip the upload step if there's no associated user ID, which
-    // is the case for CI test recordings.
-    if (recordingInitialized === false && !isTest() && ownerId) {
-      return <Upload />;
-    } else {
-      return <DevTools />;
-    }
-  } else {
-    return <Account />;
-  }
-}
-
-const connector = connect(null, { setExpectedError, setWorkspaceId });
-type PropsFromRedux = ConnectedProps<typeof connector>;
-const ConnectedPageSwitch = connector(PageSwitch);
-
-const AppRouting = () => (
-  <tokenManager.Auth0Provider>
-    <ApolloWrapper>
-      <IntercomProvider appId={"k7f741xx"}>
-        <Provider store={window.store}>
-          <App>
+const AppRouting = () => {
+  return (
+    <tokenManager.Auth0Provider>
+      <ApolloWrapper>
+        <IntercomProvider appId={"k7f741xx"}>
+          <Provider store={window.store}>
             <React.Suspense fallback={<div>Loading</div>}>
-              <ConnectedPageSwitch />
+              <App>
```

Comment:
I kept `<App>` outside of `<React.Suspense>` because I didn't want the setup work done in the `App` component to be repeated. Any particular reason why you changed how these components are nested?

Reply:
Just experimenting. Agree it should be outside the suspense.